### PR TITLE
Mithril 2 modal fixes

### PR DESCRIPTION
**Pull request by askvortsov1**
This PR:

- Reverts the hacky #2263 
- Makes `this.hide` call the global `modal.close`, fixing #2303 
- Uses `onbeforeremove` for closing animation instead of `onremove`, and only when we actually need to close the modal
- Ensures that `readyCallback` is called on modals opened from other modals.

### CODE REVIEW

File: js/src/common/components/ModalManager.js [hunk 1]
```diff
@@ -34,6 +35,13 @@ export default class ModalManager extends Component {
   animateShow(readyCallback) {
     const dismissible = !!this.attrs.state.modal.componentClass.isDismissible;
 
+    // If we are opening this modal while another modal is already open,
+    // the shown event will not run, because the modal is already open.
+    // So, we need to manually trigger the readyCallback.
+    if (this.$().hasClass('in')) {
+      readyCallback();
```
Comment (franzliedke) [inline js/src/common/components/ModalManager.js:42]: Should this also have an early return?

File: js/src/common/components/Modal.js [hunk 2]
```diff
@@ -103,7 +112,7 @@ export default class Modal extends Component {
    * Hide the modal.
    */
   hide() {
-    this.attrs.onhide();
+    this.attrs.state.close();
```
Comment (franzliedke) [inline js/src/common/components/Modal.js:115]: The naming and usage of `hide`, `onhide` and `close` is now very irritating, IMO.

File: js/src/common/components/ModalManager.js (hunk 1 above)
Comment (askvortsov1) [reply js/src/common/components/ModalManager.js:42]: Good point!

File: js/src/common/components/Modal.js (hunk 2 above)
Comment (askvortsov1) [reply js/src/common/components/Modal.js:115]: Agree, but not that much we can do. Renamed onhide to animateHide

**Review by clarkwinkelmann — APPROVED** [review]

File: js/src/common/components/Modal.js (hunk 2 above)
Comment (clarkwinkelmann) [reply js/src/common/components/Modal.js:115]: Shouldn't `onshow` also be renamed in that case? It looks inconsistent in `ModalManager` line 17 now.

**Review by datitisev — APPROVED** [review]
Seems a bit hacky, but the modal manager was bound to be 🤔

**Pull request merged**